inject-java/src/test/java/io/micronaut/aop/JdkRuntimeProxy.java

```diff
@@ -0,0 +1,87 @@
+package io.micronaut.aop;
+
+import io.micronaut.aop.chain.MethodInterceptorChain;
+import io.micronaut.aop.runtime.RuntimeProxyCreator;
+import io.micronaut.aop.runtime.RuntimeProxyDefinition;
+import io.micronaut.inject.ExecutableMethod;
+import jakarta.inject.Singleton;
+import org.jspecify.annotations.NullMarked;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+import java.util.HashMap;
+import java.util.Map;
+
+@Singleton
+@NullMarked
+public class JdkRuntimeProxy implements RuntimeProxyCreator {
```

Comment:
would be good to add a test Groovy implementation that returns the original instance and uses runtime Groovy meta-programming to replace the method and proxy instead. This will further validate what we are trying to achieve. See https://groovy-lang.org/metaprogramming.html#metaprogramming_emc

Reply:
Added